darwin/dataset/remote_dataset.py

```diff
@@ -317,25 +317,27 @@ def remove_remote(self):
         """Archives (soft-deletion) the remote dataset"""
         self.client.put(f"datasets/{self.dataset_id}/archive", payload={}, team=self.team)
 
-    def fetch_remote_files(self, filters: Optional[dict] = None):
+    def fetch_remote_files(self, parameters: Optional[dict] = None):
```

Comment:
This could be a breaking change for any clients who currently invoke this function with "named parameters" (this might not be the correct Python naming standard)

We could add an additional parameter instead; so we end up with ```filters: Optional[dict] = None, sort: ....)

Reply:
If we go by this rule, then we lock ourselves from ever changing the names of parameters ever again, but client's _might_ be using our functions with named parameters. 

The project is still in alpha, I don't see this as an issue. Clients should expect changes, that's what we defined.

Reply:
I tend to agree with @Fl4m3Ph03n1x. The argument is defined as positional, the project is not stable, I don't see argument name consistency as a major issue at this stage.

With that said, I like @myFree's suggestion about using an additional argument for sorting parameters, as the change in the `filters` name was made to make the code more readable/maintainable, and I reckon that suggestion is also pointing in that direction

Reply:
What I would like is to do is not to change the positional parameters (usually the first 1-3 args), if there are parameters that are optional or might change we first add `*` in the function header like this:

```python
def example_function(filter, *, batch_size): 
```

This was batch_size can only be called as a named parameter. 

if batch_size is no longer needed, we can first keep accepting it but give out a warning, and then later drop it. 

Sometimes we do big changes and it doesn't make sense to be backwards compatible, and in those situations we can break function signatures, names etc. But it's always nice if we can avoid them when possible. 


In this specific function, I don't exactly see the benefit of merging the filters and sort into the same argument to the function, doesn't that make things less clear and less typed compared to 

```python
def get_remote_files(self, filters: Optional[dict] = None, sort: Optional[str] = None):
    pass

# .... 

dataset.get_remote_files(sort="updated:desc") 
```

thoughts?  (we could introduce a Sort class that can parse sort strings)